test/clients/services_manager_test.rb

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env rspec
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative '../test_helper'
+
+require "yast"
+require "services-manager/clients/services_manager"
+
+describe Y2ServicesManager::Clients::ServicesManager do
+  include Yast::UIShortcuts
+
+  # Finds in the widgets tree a widget with the given id
+  #
+  # @param tree [Yast::Term]
+  # @param id [Symbol]
+  #
+  # @return [Boolean]
+  def exist_widget?(tree, id)
+    return false unless tree.is_a?(Yast::Term)
+    return true if tree.value == :id && tree.params.first == id
+
+    tree.params.any? { |t| exist_widget?(t, id) }
+  end
+
+  subject { described_class.new }
+
+  before do
+    # Mock opening and closing the dialog
+    allow(Yast::Wizard).to receive(:CreateDialog).and_return(true)
+    allow(Yast::Wizard).to receive(:CloseDialog).and_return(true)
+
+    allow(Yast::UI).to receive(:GetDisplayInfo).and_return({})
+  end
+
+  describe "#run" do
+    before do
+      allow(Yast::UI).to receive(:UserInput).and_return(*user_input)
+      allow(Yast::Popup).to receive(:ReallyAbort).and_return(true)
+
+      allow(Yast::SystemdTarget).to receive(:get_default).and_return(default_target)
+      allow(Yast::SystemdTarget).to receive(:all).and_return(tagets)
+
+      allow(Yast::PackageSystem).to receive(:PackageInstalled)
+        .with("yast2-journal").and_return(journal_installed)
+
+      allow_any_instance_of(Yast::ServicesManagerServiceClass::ServiceLoader)
```

Comment:
I am not a big fan of `allow_any_instance_of`. I rather prefer using instance doubles:

```ruby
let(:loader) do
  instance_double(Yast::ServicesManagerServiceClass::ServiceLoader, list_units: list_units)
end

before do
  allow(Yast::ServicesManagerServiceClass::ServiceLoader).to receive(:new).and_return(loader)
end
```
Or, if you prefer, you can use `allow` on the double like the [Rubocop example](https://www.rubydoc.info/gems/rubocop-rspec/1.6.0/RuboCop/Cop/RSpec/AnyInstance).

See [RSpec official documentation](https://relishapp.com/rspec/rspec-mocks/v/3-7/docs/working-with-legacy-code/any-instance) to find out more information.